### PR TITLE
Clean numerous warning that arises in hw-app-eth making it hard to read errors

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

- [provideERC20TokenInformation is not deprecated in stark context](https://github.com/LedgerHQ/ledger-live/commit/c0116b9d35ef1984e2e934101fc0719fc6f74b21) 
- [introduce signTxWithResolution helper that do explicit resolution](https://github.com/LedgerHQ/ledger-live/commit/e142c752a515f0e03402805cbc59c83647daa4e9)

### ❓ Context

- **Impacted projects**: `n/a (only tests)` 
- **Linked resource(s)**: `n/a` 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

The CI was polluted of this warns

Screenshot 2022-08-09 at 11 10 29

this PR addresses the deprecated topic to make sure the tests are up to date with the new approach.

and now it is clean again:

Screenshot 2022-08-09 at 11 18 13

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledgerjs/packages/hw-app-eth/src/Eth.ts [hunk 1]
```diff
@@ -1205,9 +1205,6 @@ export default class Eth {
   }
 
   provideERC20TokenInformation({ data }: { data: Buffer }): Promise<boolean> {
-    console.warn(
-      "hw-app-eth: eth.provideERC20TokenInformation is deprecated. signTransaction solves this for you when providing it in `resolution`."
-    );
```
Comment (haammar-ledger) [inline libs/ledgerjs/packages/hw-app-eth/src/Eth.ts:1210]: Should be replaced with just a comment maybe? So info isn't lost

Comment (gre) [reply libs/ledgerjs/packages/hw-app-eth/src/Eth.ts:1210]: what do you mean? To me this was accidentally made deprecated, it's actually legit API in the contract, e.g. for stark usage?

Comment (haammar-ledger) [reply libs/ledgerjs/packages/hw-app-eth/src/Eth.ts:1210]: Ah ok, wondering why it was logged as deprecated, but if that's not true then it should be removed indeed.

**Review by haammar-ledger — APPROVED** [review]

File: libs/ledgerjs/packages/hw-app-eth/src/Eth.ts (hunk 1 above)
Comment (gre) [reply libs/ledgerjs/packages/hw-app-eth/src/Eth.ts:1210]: when i introduced the resolve mechanism, i have wrote this deprecating assuming that all external usage of this method would no longer be needed but i didn't see the stark usage back at the time

**Pull request merged**